docs/data-tunnel/04-tunnel-messages.md

```diff
@@ -16,34 +56,84 @@ message MsgActivateTunnel {
+If a tunnel creator wants to temporarily halt packet production, they can broadcast a `MsgDeactivateTunnel` message to BandChain. This action will prevent the tunnel from sending further packets. To resume packet production, the creator can later broadcast a `MsgActivateTunnel` message.
 
 ```protobuf
-message MsgManualTriggerTunnel {
+message MsgDeactivateTunnel {
   option (cosmos.msg.v1.signer) = "creator";
-  option (amino.name)           = "tunnel/MsgManualTriggerTunnel";
+  option (amino.name)           = "tunnel/MsgDeactivateTunnel";
 
-  // tunnel_id is the ID of the tunnel to manually trigger.
-  uint64 id = 1 [(gogoproto.customname) = "TunnelID"];
+  // tunnel_id is the ID of the tunnel to deactivate.
+  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
   // creator is the address of the creator.
   string creator = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
 }
 ```
 
-## Remove Tunnel
+## Trigger Tunnel
 
-The creator can broadcast `MsgRemoveTunnel` to sunset the tunnel. Doing so will stop further data deliveries and return the remaining balance in the tunnel’s deposit address to the creator.
+While a tunnel triggers automatically based on its interval and price deviation specification, the tunnel's creator may manually trigger the tunnel's data delivery by broadcasting `MsgTriggerTunnel`.
 
 ```protobuf
-message MsgManualRemoveTunnel {
+message MsgTriggerTunnel {
   option (cosmos.msg.v1.signer) = "creator";
-  option (amino.name)           = "tunnel/MsgRemoveTunnel";
+  option (amino.name)           = "tunnel/MsgTriggerTunnel";
 
   // tunnel_id is the ID of the tunnel to manually trigger.
-  uint64 id = 1 [(gogoproto.customname) = "TunnelID"];
+  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
   // creator is the address of the creator.
   string creator = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
 }
 ```
+
+## Deposit To Tunnel
+
+Anyone can contribute funds to an existing tunnel by broadcasting a `MsgDepositToTunnel` message. This allows the tunnel to reach the required minimum deposit threshold. If needed, funds can also be withdrawn using a `MsgWithdrawFromTunnel` message.
+
+```protobuf
+message MsgDepositToTunnel {
+  option (cosmos.msg.v1.signer) = "depositor";
+  option (amino.name)           = "tunnel/MsgDepositToTunnel";
+
+  // tunnel_id defines the unique id of the tunnel.
+  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
+
+  // amount to be deposited by depositor.
+  repeated cosmos.base.v1beta1.Coin amount = 2 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (amino.dont_omitempty)   = true
+  ];
+
+  // depositor defines the deposit addresses from the tunnel.
+  string depositor = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+```
+
+## **Withdraw from Tunnel**
```

Comment:
Should we align with "Deposit to Tunnel" Section? (IMO should not bold on the header tag na)